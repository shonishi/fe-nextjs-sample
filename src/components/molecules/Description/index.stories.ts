import type { Meta, StoryObj } from '@storybook/react';
import Description from '.';

const meta: Meta<typeof Description> = {
  component: Description,
};

export default meta;
type Story = StoryObj<typeof Description>;

export const Normal: Story = {
  args: {
    description: 'あひるのフィギュアです。お風呂に浮かべて楽しいひと時。',
  },
};

export const Empty: Story = {
  args: {
    description: '',
  },
};
