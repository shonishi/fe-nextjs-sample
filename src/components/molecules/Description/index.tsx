interface Props {
  description: string;
}

export default function Description({ description }: Props) {
  return (
    <div>
      <h3 className="sr-only">Description</h3>
      <div className="space-y-6">
        <p className="text-base text-gray-900">{description}</p>
      </div>
    </div>
  );
}
