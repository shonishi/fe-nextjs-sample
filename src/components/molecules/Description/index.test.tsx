import Description from '.';
import renderer from 'react-test-renderer';

describe('Description', () => {
  it('descriptionに文字列が設定されている場合、それを元に表示されること', async () => {
    const tree = renderer
      .create(
        <Description
          {...{
            description:
              'あひるのフィギュアです。お風呂に浮かべて楽しいひと時。',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('descriptionが空の場合、空で表示されること', async () => {
    const tree = renderer
      .create(
        <Description
          {...{
            description: '',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
