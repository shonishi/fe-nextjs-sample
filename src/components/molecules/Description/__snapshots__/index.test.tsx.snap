// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description descriptionが空の場合、空で表示されること 1`] = `
<div>
  <h3
    className="sr-only"
  >
    Description
  </h3>
  <div
    className="space-y-6"
  >
    <p
      className="text-base text-gray-900"
    />
  </div>
</div>
`;

exports[`Description descriptionに文字列が設定されている場合、それを元に表示されること 1`] = `
<div>
  <h3
    className="sr-only"
  >
    Description
  </h3>
  <div
    className="space-y-6"
  >
    <p
      className="text-base text-gray-900"
    >
      あひるのフィギュアです。お風呂に浮かべて楽しいひと時。
    </p>
  </div>
</div>
`;
